cmd/start/model.go

```diff
@@ -86,50 +104,137 @@ func (m Model) executeCurrentStep() tea.Cmd {
 	currentStep := m.currentStep()
 
 	return func() tea.Msg {
-		return currentStep.fn()
+		return currentStep.fn(&m)
 	}
 }
 
+func (m Model) executeNextStep() (Model, tea.Cmd) {
+	// Check if there are more steps
+	if m.current >= len(m.steps)-1 {
+		// No more steps, we're done
+		m.done = true
+		return m, tea.Quit
+	}
+
+	// Move to next step and execute it
+	m.current++
+
+	return m, m.executeCurrentStep()
+}
+
 func (m Model) currentStep() step {
 	return m.steps[m.current]
 }
 
+func (m Model) execQuickstartScript() tea.Cmd {
+	cmd := exec.Command(m.quickstartScriptPath)
+
+	return tea.ExecProcess(cmd, func(_ error) tea.Msg {
+		return scriptCompleteMsg{}
+	})
+}
+
 func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 	switch msg := msg.(type) {
 	case tea.KeyMsg:
+		return m.handleKey(msg)
+
+	case stepCompleteMsg:
+		return m.handleStepComplete(msg)
+
+	case stepErrorMsg:
+		m.err = msg.err
+		m.quitting = true
+		// Don't quit immediately - wait for user to see the error and press a key
+		return m, nil
+
+	case scriptCompleteMsg:
+		// Script execution completed, quit the program
+		return m, tea.Quit
+	}
+
+	return m, nil
+}
+
+func (m Model) handleKey(msg tea.KeyMsg) (tea.Model, tea.Cmd) {
+	// If there's an error, any key press quits
+	if m.err != nil {
+		return m, tea.Quit
+	}
+
+	// If we're waiting for user confirmation to execute the script
+	if m.waitingToExecute {
 		switch msg.String() {
-		case "q", "esc", "ctrl+c":
+		case "y", "Y", "enter":
+			// Punch it, Chewie!
+			m.waitingToExecute = false
+			m.stepCompleteMessage = ""
+
+			if m.quickstartScriptPath != "" {
+				return m, m.execQuickstartScript()
+			}
+
+			return m.executeNextStep()
+		case "n", "N", "q", "esc":
+			// Just hang on. Hang on, Dak.
 			m.quitting = true
 			return m, tea.Quit
 		}
+		// Ignore other keys when waiting
+		return m, nil
+	}
 
-	case stepCompleteMsg:
-		// See if there's a next step, and move to it
-		if m.current < len(m.steps)-1 {
-			m.current++
-			return m, m.executeCurrentStep()
-		}
+	// Normal key handling when not waiting
+	switch msg.String() {
+	case "q", "esc":
+		m.quitting = true
+		return m, tea.Quit
+	}
 
-		// No more steps, we're done
-		m.done = true
+	return m, nil
+}
 
-		return m, tea.Quit
+func (m Model) handleStepComplete(msg stepCompleteMsg) (tea.Model, tea.Cmd) {
+	// Store any message from the completed step
+	if msg.message != "" {
+		m.stepCompleteMessage = msg.message
+	}
 
-	case stepErrorMsg:
-		m.err = msg.err
-		m.quitting = true
+	// Store quickstart script path if provided
+	if msg.quickstartScriptPath != "" {
+		m.quickstartScriptPath = msg.quickstartScriptPath
+	}
+
+	// If this step requires executing a script, do it now
```

Comment:
I like these comments - how do we feel about more liberal usage of DEBUG level logging in this repo.?

As in, maybe these comments could also be debug statements?